graphql_compiler/macros/macro_expansion.py

```diff
@@ -30,7 +31,26 @@ def _merge_non_overlapping_dicts(merge_target, new_data):
 
 
 def _merge_selection_sets(selection_set_a, selection_set_b):
-    """Merge selection sets, merging directives on name conflict."""
+    """Merge selection sets, merging directives on name conflict.
+
+    Create a selection set that contains the selections of both inputs. If there is a name
+    collision on a property field, we take the directives from both inputs on that field and
+    merge them. We disallow name collision on a vertex field.
+
+    The value None represents an empty SelectionSet.
+
+    The order of selections in the resulting SelectionSet has the following properties:
+    - property fields are before vertex fields.
+    - fields present in selection_set_b are after all other fields.
```

Comment:
presumably this only applies to vertex fields from B, right?

Reply:
Applies to both vertex fields and property fields, but separately. Think of it as "It applies to everything, but it has lower priority than the first bullet point".